tap_eip712_message/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023-, Semiotic AI, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! # EIP712 message and signature
+//!
+//! This module contains the `EIP712SignedMessage` struct which is used to sign and verify messages
+//! using EIP712 standard.
+//!
+//! # Example
+//! ```rust
+//! # use alloy::{dyn_abi::Eip712Domain, primitives::Address, signers::local::PrivateKeySigner};
+//! # let domain_separator = Eip712Domain::default();
+//! use tap_eip712_message::EIP712SignedMessage;
+//! # let wallet = PrivateKeySigner::random();
+//! # let wallet_address = wallet.address();
+//! # let message = msg::Receipt::new(Address::from([0x11u8; 20]), 100).unwrap();
+//!
+//! let signed_message = EIP712SignedMessage::new(&domain_separator, message, &wallet).unwrap();
+//! let signer = signed_message.recover_signer(&domain_separator).unwrap();
+//!
+//! assert_eq!(signer, wallet_address);
+//! ```
+//!
+
+use alloy::{
+    dyn_abi::Eip712Domain,
+    primitives::{Address, PrimitiveSignature as Signature},
+    signers::{local::PrivateKeySigner, SignerSync},
+    sol_types::SolStruct,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Eip712Error {
+    /// `alloy` wallet error
+    #[error(transparent)]
+    WalletError(#[from] alloy::signers::Error),
+
+    /// `alloy` wallet error
+    #[error(transparent)]
+    SignatureError(#[from] alloy::primitives::SignatureError),
+}
+
+/// EIP712 signed message
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct EIP712SignedMessage<M: SolStruct> {
+    /// Message to be signed
+    pub message: M,
+    /// ECDSA Signature of eip712 hash of message
+    pub signature: Signature,
+}
+
```

Comment:
Did you want to doc these? Fine by me if not but you seem to be covering a lot.

Reply:
I have a little documentation set for SignatureBytes in #269